shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java

```diff
@@ -791,31 +852,60 @@ public void autofill(SparseArray<AutofillValue> values) {
       return;
     }
 
-    final HashMap<String, TextInputChannel.TextEditState> editingValues = new HashMap<>();
-    for (int i = 0; i < values.size(); i++) {
-      int virtualId = values.keyAt(i);
+    if (configuration.enableDeltaModel) {
+      final HashMap<String, TextEditingDelta> editingDeltas = new HashMap<>();
+      for (int i = 0; i < values.size(); i++) {
+        int virtualId = values.keyAt(i);
 
-      final TextInputChannel.Configuration config = mAutofillConfigurations.get(virtualId);
-      if (config == null || config.autofill == null) {
-        continue;
+        final TextInputChannel.Configuration config = mAutofillConfigurations.get(virtualId);
+        if (config == null || config.autofill == null) {
+          continue;
+        }
+
+        final TextInputChannel.Configuration.Autofill autofill = config.autofill;
+        final String value = values.valueAt(i).getTextValue().toString();
+        final TextInputChannel.TextEditState newState =
+            new TextInputChannel.TextEditState(value, value.length(), value.length(), -1, -1);
+        final TextEditingDelta newDelta =
+            new TextEditingDelta(
+                "", value, 0, 0, value, 0, value.length(), value.length(), value.length(), -1, -1);
+
+        if (autofill.uniqueIdentifier.equals(currentAutofill.uniqueIdentifier)) {
+          // Autofilling the current client is the same as handling user input
+          // from the virtual keyboard. Setting the editable to newState and an
```

Comment:
By "virtual keyboard" do you mean the usual Android on-screen keyboard?  I think you should refer to it as the "soft keyboard" based on my understanding of Flutter's usual terminology, unless you've seen this way elsewhere.  In the framework there is a PR referring to virtual keyboards as keyboards implemented in-app in Flutter itself (https://github.com/flutter/flutter/issues/68988), so that made me notice this.

Reply:
Eh I guess we do have `virtualId` here...  I guess the naming is up to you, whatever you think is best.

Reply:
The `virtualId` here is the id of the virtual view structure used for autofill.